sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/FixedLengthRowBasedKeyValueBatch.java

```diff
@@ -46,6 +46,8 @@ private long getKeyOffsetForFixedLengthRecords(int rowId) {
   public UnsafeRow appendRow(Object kbase, long koff, int klen,
                              Object vbase, long voff, int vlen) {
     // if run out of max supported rows or page size, return null
+    assert(vlen == this.vlen);
```

Comment:
assert or require?
Like is is something that user input can never cause (if there is no bug)? assert
But otherwise this won't be enabled at runtime.

Reply:
Assert, as user input should not be able to cause this (if there is no bug).